evm-contracts/src/v0.7/dev/Operator.sol

```diff
@@ -156,6 +165,13 @@ contract Operator is
     // callback(addr+functionId) as it is untrusted.
     // See: https://solidity.readthedocs.io/en/develop/security-considerations.html#use-the-checks-effects-interactions-pattern
     (bool success, ) = callbackAddress.call(abi.encodeWithSelector(callbackFunctionId, requestId, data)); // solhint-disable-line avoid-low-level-calls
+    emit OracleResponse(
```

Comment:
Best practice is to follow the [checks, effects, interactions pattern](https://solidity.readthedocs.io/en/develop/security-considerations.html#use-the-checks-effects-interactions-pattern). This event would fall under effects. By putting it before the callback, we insulate the Operator execution from the results of the callback. If the callback runs out of gas we still emit the event. I'm a bit surprised this wasn't caught by a test, we definitely used to have one for that case.

We deviate a little bit from the CEI pattern, and have the require statement just before the callback to make sure we give the consumer enough gas. To preserve that check, I it would be best to emit the event before that check.